drivers/usb/device/usb_dc_stm32.c

```diff
@@ -963,6 +963,8 @@ void HAL_PCD_SetupStageCallback(PCD_HandleTypeDef *hpcd)
 	LOG_DBG("");
 
 	ep_state = usb_dc_stm32_get_ep_state(EP0_OUT); /* can't fail for ep0 */
+	__ASSERT(ep_state, "No corresponding ep_state for EP0");
```

Comment:
how does this fix the problem? if asserts are disabled we are back to square 1. Maybe I am missing something.

Reply:
indeed it is disabled by default, then why many drivers use it in the same way?

Reply:
> how does this fix the problem? if asserts are disabled we are back to square 1. Maybe I am missing something.

Yes, I think this is bug with assert in Zephyr, in this situation we probably need some way to print error and panic and I do not know other way of doing this. 
PS: I actually think it "fixes" Coverity.

Reply:
`k_panic()` or `k_oops()` are an alternative

Reply:
@nashif this is a consistent pattern in  Zephyr. Instead of checking a return value, we rely on asserts and expect the user to run the code with asserts enabled in order to check for those cases.